### DISCUSSION TITLE
Real time monitoring

### DISCUSSION THREAD

**Discussion by oenu** [openai/whisper#363] [Q&A] [answered]
Hi all, developer on a project that uses Whisper

We are trying to use node to monitor the python output from the cli and are unable to get the real time output from python. ie, the only data packets are sent via stdout at the end of the execution. As someone without much python experience, is this a limitation of our implementation or something that requires a custom implementation of whisper with ipc messaging bolted on?

Really excited to see where this project goes!

**Comment by jianfch** [comment, marked as answer]
To print out the tokens as they are being decode, you can add a line after and before this [line](https://github.com/openai/whisper/blob/7f3e408e092e73d472036ae2e3fba1e7c68ca4e6/whisper/decoding.py#L602):
```python
  prev_token_len = tokens.shape[-1] #add this
  # expand the tokens tensor with the selected next tokens
  tokens, completed = self.decoder.update(tokens, logits, sum_logprobs)
  print(self.tokenizer.decode(tokens[0, prev_token_len:])) #add this
```
This does not print the timestamps, you'll need additional lines to decode the timestamps tokens separately.

**Reply by oenu** [reply]
Many thanks!

**Comment by turnkit** [comment]
Would this be a good verbose option to monitor in stdout?  Seems very useful, even if just to help educate the users about internal operation.

**Reply by Disidente** [reply]
> Would this be a good verbose option to monitor in stdout? Seems very useful, even if just to help educate the users about internal operation.

Console scripting is enabled in Asynchronous mode. 
But I don't know what the problem is with Python and the stdOut that doesn't let redirect.

I have this problem. 
See here! => https://github.com/openai/whisper/discussions/538

**Reply by Disidente** [reply]
> > Would this be a good verbose option to monitor in stdout? Seems very useful, even if just to help educate the users about internal operation.
> 
> Console scripting is enabled in Asynchronous mode. But I don't know what the problem is with Python and the stdOut that doesn't let redirect.
> 
> I have this problem. See here! => #538

OK! Solved! Is a Python buffer problem!!!
To force the stdout and stderr streams to send the output of a Python script straight to a terminal (or a log file), you can set the PYTHONUNBUFFERED environment variable (to any value different from 0) or execute the python command with the -u option.

[https://www.shellhacks.com/python-script-not-showing-output-solved/](url)
